### PR TITLE
Streamline "duplicate" and "restart" routes

### PR DESCRIPTION
**Pull request by Martchus**
* Drop the "job_duplicate" event; only emit "job_restart" events
* Keep the slightly different behavior of "restart" and "duplicate" in
  place for compatibility
* Point out the difference between the "duplicate" and "restart" route
* Point out that the "dup_type_auto" parameter has only informal purposes
* Support all the options in both routes
* Allow passing zero values to flag parameters in order to disable defaults
* Share the same implementation
* Improve returning results of "job_restart" (at some point returning an
  array becomes unhandy)
* Add explicit test for the "duplicate" route
* See https://progress.opensuse.org/issues/69997

### CODE REVIEW

**Comment by Martchus** [comment]
> Hm, how did you manage to add more code then deleting while reducing duplication?

The "duplicate" route wasn't that much code. But I added an explicit test for it, added documentation for its parameters and an explanation between the difference of "duplicate" and "restart". And after all I retained compatibility - so I moved some of the parameters which were unique to "duplicate" into "restart". But when I start improving the error handling, I now don't need to care about two routes but only one anymore.

> I was never a fan of the word "restart" which by many is misunderstood as "the same job runs again" which also just recently showed up in a discussion you were also involved in. Maybe we should replace all fitting occurrences of "restart" with "clone" that we also use?

Ok. The names of the API routes are fix but we could change the UI. "clone" conflicts with the "clone-job" script which does not create clones in the openQA sense. But yes, likely it is still less confusing then restart. I guess I'll take out my renaming changes for the worker code out for now and we can think about changing any names later.

**Comment by kalikiana** [comment]
> > Hm, how did you manage to add more code then deleting while reducing duplication?
> 
> The "duplicate" route wasn't that much code. But I added an explicit test for it, added documentation for its parameters and an explanation between the difference of "duplicate" and "restart". And after all I retained compatibility - so I moved some of the parameters which were unique to "duplicate" into "restart". But when I start improving the error handling, I now don't need to care about two routes but only one anymore.
> 
> > I was never a fan of the word "restart" which by many is misunderstood as "the same job runs again" which also just recently showed up in a discussion you were also involved in. Maybe we should replace all fitting occurrences of "restart" with "clone" that we also use?
> 
> Ok. The names of the API routes are fix but we could change the UI. "clone" conflicts with the "clone-job" script which does not create clones in the openQA sense. But yes, likely it is still less confusing then restart. I guess I'll take out my renaming changes for the worker code out for now and we can think about changing any names later.

I feel like we should actually be more explicit in the UX going forward. Any term can be misunderstood and since we overload terminology based on context I don't blame users for getting confused. This is the same as with uses of "test" or "test results". In that sense I don't mind so much if we use "restart" or "duplicate" or "clone" so long as we add a little note about what it will do. (and yes, not in the PR of course)

File: lib/OpenQA/WebAPI/Controller/API/V1/Job.pm [hunk 1]
```diff
@@ -680,65 +680,97 @@ sub done {
...
-
-=cut
-
-sub restart {
-    my ($self) = @_;
+sub _restart {
+    my ($self, %args) = @_;
 
-    my @flags      = qw(force skip_parents skip_children skip_ok_result_children);
+    my $dup_route  = $args{duplicate_route_compatibility};
```
Comment (kalikiana) [inline lib/OpenQA/WebAPI/Controller/API/V1/Job.pm:686]: I would prefer setting the real args here i.e. `skip_aborting_jobs`, `force` rather than special-casing this

Comment (Martchus) [reply lib/OpenQA/WebAPI/Controller/API/V1/Job.pm:686]: But this is just about the defaults - the "duplicate" route now supports all parameters the "restart" route does (and vice versa). I don't want to duplicate code for the argument validation so I thought it is best to pass a single flag here. Also note that this argument slightly changes the format of the resulting array.
